engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigratorImpl.java

```diff
@@ -143,8 +144,13 @@ private void runMigration(final MigrationTask migrationTask, final int index, fi
     migrationTask.runMigration(processingState);
     final var duration = System.currentTimeMillis() - startTime;
 
+    markMigrationFinished(migrationTask.getIdentifier());
```

Comment:
🔧 `handleMigrationTask` would be easier to understand if we moved this invocation to that method. Note that it is also invoked from the else branch there. So, then it's clear that it is invoked from both branches in that method.